packages/e2e-tests/helpers.ts

```diff
@@ -315,11 +315,23 @@ export async function closeSource(screen: Screen, url: string) {
 export async function selectSource(screen: Screen, url: string) {
   debugPrint(`Opening source "${chalk.bold(url)}"`, "selectSource");
 
+  // If the source is already open, just focus it.
+  const sourceTab = screen.locator(`[data-test-name="Source-${url}"]`);
+  if (await sourceTab.isVisible()) {
+    debugPrint(`Source "${chalk.bold(url)}" already open`, "selectSource");
+    await sourceTab.locator("button").click();
+    return;
+  }
+
+  // Otherwise find it in the sources tree.
   const pane = await getSourcesPane(screen);
 
   let foundSource = false;
 
   while (true) {
+    // Wait for the panel's HTML to settle (to reflect the new toggled state.)
+    await pane.innerHTML();
```

Comment:
This seems to have been the issue that was breaking Mark's tests. I guess we weren't waiting for the HTML to update between opening one source and the next 😅